File: testData/petShopRoutes.py
from flask import Flask, jsonify, request, abort

app = Flask(__name__)

pets = [
    {'id': 1, 'name': 'Rex', 'type': 'Dog', 'age': 5},
    {'id': 2, 'name': 'Mittens', 'type': 'Cat', 'age': 3}
]

@app.route('/pets', methods=['GET'])
def get_pets():
    return jsonify(pets)

@app.route('/pets/<int:pet_id>', methods=['GET'])
def get_pet(pet_id):
    pet = next((pet for pet in pets if pet['id'] == pet_id), None)
    if pet is None:
        abort(404)
    return jsonify(pet)

@app.route('/pets', methods=['POST'])
def create_pet():
    if not request.json or not 'name' in request.json:
        abort(400)
    pet = {
        'id': pets[-1]['id'] + 1 if pets else 1,
        'name': request.json['name'],
        'type': request.json.get('type', ""),
        'age': request.json.get('age', 0)
    }
    pets.append(pet)
    return jsonify(pet), 201

@app.route('/pets/<int:pet_id>', methods=['PUT'])
def update_pet(pet_id):
    pet = next((pet for pet in pets if pet['id'] == pet_id), None)
    if pet is None:
        abort(404)
    if not request.json:
        abort(400)
    pet['name'] = request.json.get('name', pet['name'])
    pet['type'] = request.json.get('type', pet['type'])
    pet['age'] = request.json.get('age', pet['age'])
    return jsonify(pet)

@app.route('/pets/<int:pet_id>', methods=['DELETE'])
def delete_pet(pet_id):
    global pets
    pet = next((pet for pet in pets if pet['id'] == pet_id), None)
    if pet is None:
        abort(404)
    pets = [pet for pet in pets if pet['id'] != pet_id]
    return jsonify({'result': True})

if __name__ == '__main__':
    app.run(debug=True)

File: testData/petShopRoutes.php
header("Content-Type: application/json");

// Simulated database of pets
$pets = [
    ['id' => 1, 'name' => 'Rex', 'type' => 'Dog', 'age' => 5],
    ['id' => 2, 'name' => 'Mittens', 'type' => 'Cat', 'age' => 3]
];

// Utility function to find a pet by ID
function findPetById($pets, $id) {
    foreach ($pets as $pet) {
        if ($pet['id'] == $id) {
            return $pet;
        }
    }
    return null;
}

$method = $_SERVER['REQUEST_METHOD'];
$path = explode('/', trim($_SERVER['PATH_INFO'],'/'));
$petId = isset($path[1]) ? (int)$path[1] : null;

switch ($method) {
    case 'GET':
        if ($petId) {
            $pet = findPetById($pets, $petId);
            if ($pet) {
                echo json_encode($pet);
            } else {
                http_response_code(404);
                echo json_encode(['error' => 'Pet not found']);
            }
        } else {
            echo json_encode($pets);
        }
        break;
    case 'POST':
        $input = json_decode(file_get_contents('php://input'), true);
        if (!is_array($input) || !isset($input['name']) || !isset($input['type']) || !isset($input['age'])) {
            http_response_code(400);
            echo json_encode(['error' => 'Bad request']);
            break;
        }
        $newPet = [
            'id' => end($pets)['id'] + 1,
            'name' => $input['name'],
            'type' => $input['type'],
            'age' => $input['age']
        ];
        $pets[] = $newPet;
        echo json_encode($newPet);
        break;
    case 'PUT':
        if (!$petId || !$pet = findPetById($pets, $petId)) {
            http_response_code(404);
            echo json_encode(['error' => 'Pet not found']);
            break;
        }
        $input = json_decode(file_get_contents('php://input'), true);
        if (!is_array($input)) {
            http_response_code(400);
            echo json_encode(['error' => 'Bad request']);
            break;
        }
        // Update the pet
        foreach ($pets as &$pet) {
            if ($pet['id'] == $petId) {
                $pet['name'] = $input['name'] ?? $pet['name'];
                $pet['type'] = $input['type'] ?? $pet['type'];
                $pet['age'] = $input['age'] ?? $pet['age'];
                echo json_encode($pet);
                break;
            }
        }
        break;
    case 'DELETE':
        if (!$petId) {
            http_response_code(400);
            echo json_encode(['error' => 'Bad request']);
            break;
        }
        $found = false;
        foreach ($pets as $i => $pet) {
            if ($pet['id'] == $petId) {
                array_splice($pets, $i, 1);
                $found = true;
                break;
            }
        }
        if ($found) {
            http_response_code(204);
        } else {
            http_response_code(404);
            echo json_encode(['error' => 'Pet not found']);
        }
        break;
    default:
        http_response_code(405);
        echo json_encode(['error' => 'Method not allowed']);
        break;
}
